djangocms_version_locking/monkeypatch/admin.py

```diff
@@ -1,13 +1,42 @@
 from django.utils.translation import ugettext_lazy as _
 
-from djangocms_versioning import admin, models
+from djangocms_versioning import admin, models, constants
+
+from ..models import VersionLock
+
+
+def new_save(old_save):
+    """
+    Override the save method to add a version lock
+    """
+    def inner(version, **kwargs):
+        old_save(version, **kwargs)
+        # A draft version is locked by default
+        if version.state == constants.DRAFT:
+            # create lock
+            VersionLock.objects.create(
+                version=version,
+                created_by=version.created_by
+            )
+        # A published version has no lock, an existing lock should be removed
+        elif version.state == constants.PUBLISHED:
+            try:
```

Comment:
`VersionLock.objects.filter(version=version).delete()` and there's no need for additional checks